pkg/controller/atlasdeployment/deployment.go

```diff
@@ -133,10 +133,6 @@ func removeOutdatedFields(removeFrom *mongodbatlas.Cluster, lookAt *mongodbatlas
 		} else {
 			result.ProviderSettings.AutoScaling.Compute = &mongodbatlas.Compute{}
 		}
-
-		if lookAt.AutoScaling.DiskGBEnabled != nil && *lookAt.AutoScaling.DiskGBEnabled {
```

Comment:
Hey @igor-karpukhin,
Are you changing the whole concept of the operator should not touch disk/instance size if autoscaling is enabled?

For example, we have a deploymentA(Kubernetes object) with a 10GB disk, and disk autoscaling is enabled.
After some time, autoscaler on the Atlas platform decided to increase the disk size to 15GB, the increasing process is finished and now our atlas cluster has a 15GB disk.
What value should we expect in the Kubernetes object related to that deploymentA 10GB or 15GB?

Reply:
The API technically allows disk size to be configured with autoscaling enabled. But, I just revert it to the previous state so it's not configurable if the autoscaling is enabled.